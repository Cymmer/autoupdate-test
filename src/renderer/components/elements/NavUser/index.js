import cn from 'classnames';
import { getUser } from 'ducks';
import { useOnClickOutside } from 'hooks';
import PropTypes from 'prop-types';
import React, { useRef, useState } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Card from '../Card';
import Icon from '../Icon';
import UserImage from '../UserImage';
import styles from './styles.module.scss';

const NavUser = ({ user, className }) => {
  const [isDropdownToggled, toggleDropdown] = useState(false);

  const ref = useRef();

  useOnClickOutside(ref, () => toggleDropdown(false));

  return (
    <div ref={ref} className={cn(styles.NavUser, className)}>
      <button
        className={styles.NavUser_button}
        data-test="dropdownButton"
        id="navUserButton"
        tabIndex={0}
        type="button"
        onClick={() => toggleDropdown(!isDropdownToggled)}
      >
        <UserImage className={styles.NavUser_image} image={user.profile_pic} />
        <span className={styles.NavUser_name}>{user.last_name}</span>
        <Icon className={styles.NavUser_caret} icon="keyboard_arrow_down" />
      </button>
      <Card
        className={cn({
          [styles.NavUser_dropdown]: !isDropdownToggled,
          [styles.NavUser_dropdown___toggled]: isDropdownToggled,
        })}
      >
        {/* Note: Temporarily hidden for Offline Version */}
        {/* <Link
          className={styles.NavUser_dropdown_link}
          id="myAccountButton"
          to="/student/account/personal-information"
          onClick={() => toggleDropdown(!isDropdownToggled)}
        >
          <Icon
            className={styles.NavUser_dropdown_link_icon}
            icon="account_circle"
          />
          My Account
        </Link> */}
        <Link
          className={styles.NavUser_dropdown_link}
          id="logoutButton"
          to="/logout"
        >
          <Icon
            className={styles.NavUser_dropdown_link_icon}
            icon="exit_to_app"
          />
          Logout
        </Link>
      </Card>
    </div>
  );
};

NavUser.defaultProps = {
  className: null,
};

NavUser.propTypes = {
  className: PropTypes.string,
  user: PropTypes.oneOfType([PropTypes.object]).isRequired,
};

const mapStateToProps = (store) => ({
  user: getUser(store),
});

export default connect(mapStateToProps, null)(NavUser);
