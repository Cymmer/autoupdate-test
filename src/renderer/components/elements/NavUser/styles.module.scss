@import '../../../scss/mixins';
@import '../../../scss/variables';
@import '../../../scss/themes';

@mixin dropdown($display) {
  display: $display;
  flex-direction: column;
  position: absolute;
  top: 52px;
  right: 0;
  box-shadow: $shadow-normal;
  width: 150px;
  z-index: 1;
  border: none;

  @content;
}

.NavUser {
  position: relative;

  &_image {
    width: 36px;
    margin: 0 8px 0 0;

    @include breakpoint(sm) {
      margin: 0;
    }
  }

  &_caret {
    margin: 0 0 0 4px;
  }

  &_name {
    @include heading(0.75rem, 1.67);
  }

  &_caret,
  &_name {
    @include breakpoint(sm) {
      display: none;
    }
  }

  &_button {
    @include flex-center;
    @include ease(0.2s);
    @include theme {
      color: t('textColor-700');
    }
    border: none;
    outline: none;
    background: transparent;
    cursor: pointer;
  }

  &_dropdown {
    @include dropdown(none);

    &___toggled {
      @include dropdown(flex);
    }

    &_link {
      @include heading(0.75rem, 1.67);
      @include ease(0.3s);
      text-decoration: none;
      padding: 16px;
      display: flex;
      align-items: center;

      @include theme {
        color: t('textColor-700');

        &:hover {
          background-color: t('cardBackgroundColor--hover');
        }

        &:focus,
        &:active {
          background-color: t('cardBackgroundColor--active');
        }
      }

      &:first-child {
        border-radius: 16px 16px 0 0;
      }

      &:last-child {
        border-radius: 0 0 16px 16px;
      }

      &_icon {
        margin-right: 8px;
        @include theme {
          color: t('textColor-400');
        }
      }
    }
  }
}
