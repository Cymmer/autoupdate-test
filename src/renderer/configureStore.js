import { createHashHistory } from 'history';
import { routerMiddleware } from 'react-router-redux';
import { applyMiddleware, compose, createStore } from 'redux';
import { persistReducer } from 'redux-persist';
import hardSet from 'redux-persist/lib/stateReconciler/hardSet';
import storage from 'redux-persist/lib/storage';
import rootReducer from './ducks/index';

export const configureStore = () => {
  // create middlewares
  const history = createHashHistory();

  const composeEnhancers =
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const middleware = applyMiddleware(routerMiddleware(history));

  // create a persist version of the reducer
  const persistConfig = {
    key: 'codechum-desktop',
    storage,
    stateReconciler: hardSet,
  };
  const persistedReducer = persistReducer(persistConfig, rootReducer);

  // create store
  const store = createStore(persistedReducer, composeEnhancers(middleware));

  return store;
};
