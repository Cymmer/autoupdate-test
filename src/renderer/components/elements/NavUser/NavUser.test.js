import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { findByTestAttr } from 'codechum-app-utils';
import GLOBALS from 'codechum-app-globals';
import NavUser from './index';
import { configureStore } from '../../../configureStore';

describe('RENDERING - NavUser', () => {
  it('should render with default props', () => {
    const store = configureStore();
    const component = mount(
      <Provider store={store}>
        <BrowserRouter basename="/">
          <NavUser />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  it('should have the proper links for the user type student', () => {
    const store = configureStore();
    const component = mount(
      <Provider store={store}>
        <BrowserRouter basename="/">
          <NavUser userType={GLOBALS.USER_TYPES.STUDENT} />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  it('should have the proper links for the user type teacher', () => {
    const store = configureStore();
    const component = mount(
      <Provider store={store}>
        <BrowserRouter basename="/">
          <NavUser userType={GLOBALS.USER_TYPES.TEACHER} />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  it('should have the proper links for the user type manager', () => {
    const store = configureStore();
    const component = mount(
      <Provider store={store}>
        <BrowserRouter basename="/">
          <NavUser userType={GLOBALS.USER_TYPES.MANAGER} />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  it('should have the proper links for the user type admin', () => {
    const store = configureStore();
    const component = mount(
      <Provider store={store}>
        <BrowserRouter basename="/">
          <NavUser userType={GLOBALS.USER_TYPES.ADMIN} />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  it('should update the classes on dropdown toggled on desktop', () => {
    const store = configureStore();
    const component = mount(
      <Provider store={store}>
        <BrowserRouter basename="/">
          <NavUser />
        </BrowserRouter>
      </Provider>
    );

    findByTestAttr(component, 'dropdownButton').simulate('click');
    component.update();
    expect(component).toMatchSnapshot();
  });
});
