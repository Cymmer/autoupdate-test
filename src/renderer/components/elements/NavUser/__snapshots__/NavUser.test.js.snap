// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - NavUser should have the proper links for the user type admin 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter
    basename="/"
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "#/",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(NavUser)
        userType="A"
      >
        <NavUser
          className={null}
          dispatch={[Function]}
          user={Object {}}
          userType="A"
        >
          <div
            className="NavUser"
          >
            <button
              className="NavUser_button"
              data-test="dropdownButton"
              id="navUserButton"
              onClick={[Function]}
              tabIndex={0}
              type="button"
            >
              <UserImage
                className="NavUser_image"
                id={null}
                image={null}
                imageClassName={null}
              >
                <div
                  className="NavUser_image"
                >
                  <img
                    alt="User"
                    className="UserImage_image"
                    data-test="image"
                    id={null}
                    src="test-file-stub"
                  />
                </div>
              </UserImage>
              <span
                className="NavUser_name"
              />
              <Icon
                className="NavUser_caret"
                icon="keyboard_arrow_down"
                id={null}
                isPulsating={false}
                style={null}
              >
                <i
                  className="Icon NavUser_caret"
                  data-test="icon"
                  id={null}
                  style={null}
                >
                  keyboard_arrow_down
                </i>
              </Icon>
            </button>
            <Card
              className="NavUser_dropdown"
              id={null}
              onClick={null}
              onMouseEnter={null}
              onMouseLeave={null}
            >
              <div
                className="Card NavUser_dropdown"
                data-test="card"
                id={null}
                onClick={null}
                onKeyPress={null}
                onMouseEnter={null}
                onMouseLeave={null}
                role={null}
                tabIndex={null}
              >
                <Link
                  className="NavUser_dropdown_link"
                  id="myAccountButton"
                  onClick={[Function]}
                  to="/student/account/personal-information"
                >
                  <LinkAnchor
                    className="NavUser_dropdown_link"
                    href="/student/account/personal-information"
                    id="myAccountButton"
                    navigate={[Function]}
                    onClick={[Function]}
                  >
                    <a
                      className="NavUser_dropdown_link"
                      href="/student/account/personal-information"
                      id="myAccountButton"
                      onClick={[Function]}
                    >
                      <Icon
                        className="NavUser_dropdown_link_icon"
                        icon="account_circle"
                        id={null}
                        isPulsating={false}
                        style={null}
                      >
                        <i
                          className="Icon NavUser_dropdown_link_icon"
                          data-test="icon"
                          id={null}
                          style={null}
                        >
                          account_circle
                        </i>
                      </Icon>
                      My Account
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  className="NavUser_dropdown_link"
                  id="logoutButton"
                  to="/logout"
                >
                  <LinkAnchor
                    className="NavUser_dropdown_link"
                    href="/logout"
                    id="logoutButton"
                    navigate={[Function]}
                  >
                    <a
                      className="NavUser_dropdown_link"
                      href="/logout"
                      id="logoutButton"
                      onClick={[Function]}
                    >
                      <Icon
                        className="NavUser_dropdown_link_icon"
                        icon="exit_to_app"
                        id={null}
                        isPulsating={false}
                        style={null}
                      >
                        <i
                          className="Icon NavUser_dropdown_link_icon"
                          data-test="icon"
                          id={null}
                          style={null}
                        >
                          exit_to_app
                        </i>
                      </Icon>
                      Logout
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </Card>
          </div>
        </NavUser>
      </Connect(NavUser)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`RENDERING - NavUser should have the proper links for the user type manager 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter
    basename="/"
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "#/",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(NavUser)
        userType="M"
      >
        <NavUser
          className={null}
          dispatch={[Function]}
          user={Object {}}
          userType="M"
        >
          <div
            className="NavUser"
          >
            <button
              className="NavUser_button"
              data-test="dropdownButton"
              id="navUserButton"
              onClick={[Function]}
              tabIndex={0}
              type="button"
            >
              <UserImage
                className="NavUser_image"
                id={null}
                image={null}
                imageClassName={null}
              >
                <div
                  className="NavUser_image"
                >
                  <img
                    alt="User"
                    className="UserImage_image"
                    data-test="image"
                    id={null}
                    src="test-file-stub"
                  />
                </div>
              </UserImage>
              <span
                className="NavUser_name"
              />
              <Icon
                className="NavUser_caret"
                icon="keyboard_arrow_down"
                id={null}
                isPulsating={false}
                style={null}
              >
                <i
                  className="Icon NavUser_caret"
                  data-test="icon"
                  id={null}
                  style={null}
                >
                  keyboard_arrow_down
                </i>
              </Icon>
            </button>
            <Card
              className="NavUser_dropdown"
              id={null}
              onClick={null}
              onMouseEnter={null}
              onMouseLeave={null}
            >
              <div
                className="Card NavUser_dropdown"
                data-test="card"
                id={null}
                onClick={null}
                onKeyPress={null}
                onMouseEnter={null}
                onMouseLeave={null}
                role={null}
                tabIndex={null}
              >
                <Link
                  className="NavUser_dropdown_link"
                  id="myAccountButton"
                  onClick={[Function]}
                  to="/student/account/personal-information"
                >
                  <LinkAnchor
                    className="NavUser_dropdown_link"
                    href="/student/account/personal-information"
                    id="myAccountButton"
                    navigate={[Function]}
                    onClick={[Function]}
                  >
                    <a
                      className="NavUser_dropdown_link"
                      href="/student/account/personal-information"
                      id="myAccountButton"
                      onClick={[Function]}
                    >
                      <Icon
                        className="NavUser_dropdown_link_icon"
                        icon="account_circle"
                        id={null}
                        isPulsating={false}
                        style={null}
                      >
                        <i
                          className="Icon NavUser_dropdown_link_icon"
                          data-test="icon"
                          id={null}
                          style={null}
                        >
                          account_circle
                        </i>
                      </Icon>
                      My Account
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  className="NavUser_dropdown_link"
                  id="logoutButton"
                  to="/logout"
                >
                  <LinkAnchor
                    className="NavUser_dropdown_link"
                    href="/logout"
                    id="logoutButton"
                    navigate={[Function]}
                  >
                    <a
                      className="NavUser_dropdown_link"
                      href="/logout"
                      id="logoutButton"
                      onClick={[Function]}
                    >
                      <Icon
                        className="NavUser_dropdown_link_icon"
                        icon="exit_to_app"
                        id={null}
                        isPulsating={false}
                        style={null}
                      >
                        <i
                          className="Icon NavUser_dropdown_link_icon"
                          data-test="icon"
                          id={null}
                          style={null}
                        >
                          exit_to_app
                        </i>
                      </Icon>
                      Logout
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </Card>
          </div>
        </NavUser>
      </Connect(NavUser)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`RENDERING - NavUser should have the proper links for the user type student 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter
    basename="/"
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "#/",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(NavUser)
        userType="S"
      >
        <NavUser
          className={null}
          dispatch={[Function]}
          user={Object {}}
          userType="S"
        >
          <div
            className="NavUser"
          >
            <button
              className="NavUser_button"
              data-test="dropdownButton"
              id="navUserButton"
              onClick={[Function]}
              tabIndex={0}
              type="button"
            >
              <UserImage
                className="NavUser_image"
                id={null}
                image={null}
                imageClassName={null}
              >
                <div
                  className="NavUser_image"
                >
                  <img
                    alt="User"
                    className="UserImage_image"
                    data-test="image"
                    id={null}
                    src="test-file-stub"
                  />
                </div>
              </UserImage>
              <span
                className="NavUser_name"
              />
              <Icon
                className="NavUser_caret"
                icon="keyboard_arrow_down"
                id={null}
                isPulsating={false}
                style={null}
              >
                <i
                  className="Icon NavUser_caret"
                  data-test="icon"
                  id={null}
                  style={null}
                >
                  keyboard_arrow_down
                </i>
              </Icon>
            </button>
            <Card
              className="NavUser_dropdown"
              id={null}
              onClick={null}
              onMouseEnter={null}
              onMouseLeave={null}
            >
              <div
                className="Card NavUser_dropdown"
                data-test="card"
                id={null}
                onClick={null}
                onKeyPress={null}
                onMouseEnter={null}
                onMouseLeave={null}
                role={null}
                tabIndex={null}
              >
                <Link
                  className="NavUser_dropdown_link"
                  id="myAccountButton"
                  onClick={[Function]}
                  to="/student/account/personal-information"
                >
                  <LinkAnchor
                    className="NavUser_dropdown_link"
                    href="/student/account/personal-information"
                    id="myAccountButton"
                    navigate={[Function]}
                    onClick={[Function]}
                  >
                    <a
                      className="NavUser_dropdown_link"
                      href="/student/account/personal-information"
                      id="myAccountButton"
                      onClick={[Function]}
                    >
                      <Icon
                        className="NavUser_dropdown_link_icon"
                        icon="account_circle"
                        id={null}
                        isPulsating={false}
                        style={null}
                      >
                        <i
                          className="Icon NavUser_dropdown_link_icon"
                          data-test="icon"
                          id={null}
                          style={null}
                        >
                          account_circle
                        </i>
                      </Icon>
                      My Account
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  className="NavUser_dropdown_link"
                  id="logoutButton"
                  to="/logout"
                >
                  <LinkAnchor
                    className="NavUser_dropdown_link"
                    href="/logout"
                    id="logoutButton"
                    navigate={[Function]}
                  >
                    <a
                      className="NavUser_dropdown_link"
                      href="/logout"
                      id="logoutButton"
                      onClick={[Function]}
                    >
                      <Icon
                        className="NavUser_dropdown_link_icon"
                        icon="exit_to_app"
                        id={null}
                        isPulsating={false}
                        style={null}
                      >
                        <i
                          className="Icon NavUser_dropdown_link_icon"
                          data-test="icon"
                          id={null}
                          style={null}
                        >
                          exit_to_app
                        </i>
                      </Icon>
                      Logout
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </Card>
          </div>
        </NavUser>
      </Connect(NavUser)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`RENDERING - NavUser should have the proper links for the user type teacher 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter
    basename="/"
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "#/",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(NavUser)
        userType="T"
      >
        <NavUser
          className={null}
          dispatch={[Function]}
          user={Object {}}
          userType="T"
        >
          <div
            className="NavUser"
          >
            <button
              className="NavUser_button"
              data-test="dropdownButton"
              id="navUserButton"
              onClick={[Function]}
              tabIndex={0}
              type="button"
            >
              <UserImage
                className="NavUser_image"
                id={null}
                image={null}
                imageClassName={null}
              >
                <div
                  className="NavUser_image"
                >
                  <img
                    alt="User"
                    className="UserImage_image"
                    data-test="image"
                    id={null}
                    src="test-file-stub"
                  />
                </div>
              </UserImage>
              <span
                className="NavUser_name"
              />
              <Icon
                className="NavUser_caret"
                icon="keyboard_arrow_down"
                id={null}
                isPulsating={false}
                style={null}
              >
                <i
                  className="Icon NavUser_caret"
                  data-test="icon"
                  id={null}
                  style={null}
                >
                  keyboard_arrow_down
                </i>
              </Icon>
            </button>
            <Card
              className="NavUser_dropdown"
              id={null}
              onClick={null}
              onMouseEnter={null}
              onMouseLeave={null}
            >
              <div
                className="Card NavUser_dropdown"
                data-test="card"
                id={null}
                onClick={null}
                onKeyPress={null}
                onMouseEnter={null}
                onMouseLeave={null}
                role={null}
                tabIndex={null}
              >
                <Link
                  className="NavUser_dropdown_link"
                  id="myAccountButton"
                  onClick={[Function]}
                  to="/student/account/personal-information"
                >
                  <LinkAnchor
                    className="NavUser_dropdown_link"
                    href="/student/account/personal-information"
                    id="myAccountButton"
                    navigate={[Function]}
                    onClick={[Function]}
                  >
                    <a
                      className="NavUser_dropdown_link"
                      href="/student/account/personal-information"
                      id="myAccountButton"
                      onClick={[Function]}
                    >
                      <Icon
                        className="NavUser_dropdown_link_icon"
                        icon="account_circle"
                        id={null}
                        isPulsating={false}
                        style={null}
                      >
                        <i
                          className="Icon NavUser_dropdown_link_icon"
                          data-test="icon"
                          id={null}
                          style={null}
                        >
                          account_circle
                        </i>
                      </Icon>
                      My Account
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  className="NavUser_dropdown_link"
                  id="logoutButton"
                  to="/logout"
                >
                  <LinkAnchor
                    className="NavUser_dropdown_link"
                    href="/logout"
                    id="logoutButton"
                    navigate={[Function]}
                  >
                    <a
                      className="NavUser_dropdown_link"
                      href="/logout"
                      id="logoutButton"
                      onClick={[Function]}
                    >
                      <Icon
                        className="NavUser_dropdown_link_icon"
                        icon="exit_to_app"
                        id={null}
                        isPulsating={false}
                        style={null}
                      >
                        <i
                          className="Icon NavUser_dropdown_link_icon"
                          data-test="icon"
                          id={null}
                          style={null}
                        >
                          exit_to_app
                        </i>
                      </Icon>
                      Logout
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </Card>
          </div>
        </NavUser>
      </Connect(NavUser)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`RENDERING - NavUser should render with default props 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter
    basename="/"
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "#/",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(NavUser)>
        <NavUser
          className={null}
          dispatch={[Function]}
          user={Object {}}
        >
          <div
            className="NavUser"
          >
            <button
              className="NavUser_button"
              data-test="dropdownButton"
              id="navUserButton"
              onClick={[Function]}
              tabIndex={0}
              type="button"
            >
              <UserImage
                className="NavUser_image"
                id={null}
                image={null}
                imageClassName={null}
              >
                <div
                  className="NavUser_image"
                >
                  <img
                    alt="User"
                    className="UserImage_image"
                    data-test="image"
                    id={null}
                    src="test-file-stub"
                  />
                </div>
              </UserImage>
              <span
                className="NavUser_name"
              />
              <Icon
                className="NavUser_caret"
                icon="keyboard_arrow_down"
                id={null}
                isPulsating={false}
                style={null}
              >
                <i
                  className="Icon NavUser_caret"
                  data-test="icon"
                  id={null}
                  style={null}
                >
                  keyboard_arrow_down
                </i>
              </Icon>
            </button>
            <Card
              className="NavUser_dropdown"
              id={null}
              onClick={null}
              onMouseEnter={null}
              onMouseLeave={null}
            >
              <div
                className="Card NavUser_dropdown"
                data-test="card"
                id={null}
                onClick={null}
                onKeyPress={null}
                onMouseEnter={null}
                onMouseLeave={null}
                role={null}
                tabIndex={null}
              >
                <Link
                  className="NavUser_dropdown_link"
                  id="myAccountButton"
                  onClick={[Function]}
                  to="/student/account/personal-information"
                >
                  <LinkAnchor
                    className="NavUser_dropdown_link"
                    href="/student/account/personal-information"
                    id="myAccountButton"
                    navigate={[Function]}
                    onClick={[Function]}
                  >
                    <a
                      className="NavUser_dropdown_link"
                      href="/student/account/personal-information"
                      id="myAccountButton"
                      onClick={[Function]}
                    >
                      <Icon
                        className="NavUser_dropdown_link_icon"
                        icon="account_circle"
                        id={null}
                        isPulsating={false}
                        style={null}
                      >
                        <i
                          className="Icon NavUser_dropdown_link_icon"
                          data-test="icon"
                          id={null}
                          style={null}
                        >
                          account_circle
                        </i>
                      </Icon>
                      My Account
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  className="NavUser_dropdown_link"
                  id="logoutButton"
                  to="/logout"
                >
                  <LinkAnchor
                    className="NavUser_dropdown_link"
                    href="/logout"
                    id="logoutButton"
                    navigate={[Function]}
                  >
                    <a
                      className="NavUser_dropdown_link"
                      href="/logout"
                      id="logoutButton"
                      onClick={[Function]}
                    >
                      <Icon
                        className="NavUser_dropdown_link_icon"
                        icon="exit_to_app"
                        id={null}
                        isPulsating={false}
                        style={null}
                      >
                        <i
                          className="Icon NavUser_dropdown_link_icon"
                          data-test="icon"
                          id={null}
                          style={null}
                        >
                          exit_to_app
                        </i>
                      </Icon>
                      Logout
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </Card>
          </div>
        </NavUser>
      </Connect(NavUser)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`RENDERING - NavUser should update the classes on dropdown toggled on desktop 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter
    basename="/"
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "#/",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(NavUser)>
        <NavUser
          className={null}
          dispatch={[Function]}
          user={Object {}}
        >
          <div
            className="NavUser"
          >
            <button
              className="NavUser_button"
              data-test="dropdownButton"
              id="navUserButton"
              onClick={[Function]}
              tabIndex={0}
              type="button"
            >
              <UserImage
                className="NavUser_image"
                id={null}
                image={null}
                imageClassName={null}
              >
                <div
                  className="NavUser_image"
                >
                  <img
                    alt="User"
                    className="UserImage_image"
                    data-test="image"
                    id={null}
                    src="test-file-stub"
                  />
                </div>
              </UserImage>
              <span
                className="NavUser_name"
              />
              <Icon
                className="NavUser_caret"
                icon="keyboard_arrow_down"
                id={null}
                isPulsating={false}
                style={null}
              >
                <i
                  className="Icon NavUser_caret"
                  data-test="icon"
                  id={null}
                  style={null}
                >
                  keyboard_arrow_down
                </i>
              </Icon>
            </button>
            <Card
              className="NavUser_dropdown___toggled"
              id={null}
              onClick={null}
              onMouseEnter={null}
              onMouseLeave={null}
            >
              <div
                className="Card NavUser_dropdown___toggled"
                data-test="card"
                id={null}
                onClick={null}
                onKeyPress={null}
                onMouseEnter={null}
                onMouseLeave={null}
                role={null}
                tabIndex={null}
              >
                <Link
                  className="NavUser_dropdown_link"
                  id="myAccountButton"
                  onClick={[Function]}
                  to="/student/account/personal-information"
                >
                  <LinkAnchor
                    className="NavUser_dropdown_link"
                    href="/student/account/personal-information"
                    id="myAccountButton"
                    navigate={[Function]}
                    onClick={[Function]}
                  >
                    <a
                      className="NavUser_dropdown_link"
                      href="/student/account/personal-information"
                      id="myAccountButton"
                      onClick={[Function]}
                    >
                      <Icon
                        className="NavUser_dropdown_link_icon"
                        icon="account_circle"
                        id={null}
                        isPulsating={false}
                        style={null}
                      >
                        <i
                          className="Icon NavUser_dropdown_link_icon"
                          data-test="icon"
                          id={null}
                          style={null}
                        >
                          account_circle
                        </i>
                      </Icon>
                      My Account
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  className="NavUser_dropdown_link"
                  id="logoutButton"
                  to="/logout"
                >
                  <LinkAnchor
                    className="NavUser_dropdown_link"
                    href="/logout"
                    id="logoutButton"
                    navigate={[Function]}
                  >
                    <a
                      className="NavUser_dropdown_link"
                      href="/logout"
                      id="logoutButton"
                      onClick={[Function]}
                    >
                      <Icon
                        className="NavUser_dropdown_link_icon"
                        icon="exit_to_app"
                        id={null}
                        isPulsating={false}
                        style={null}
                      >
                        <i
                          className="Icon NavUser_dropdown_link_icon"
                          data-test="icon"
                          id={null}
                          style={null}
                        >
                          exit_to_app
                        </i>
                      </Icon>
                      Logout
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </Card>
          </div>
        </NavUser>
      </Connect(NavUser)>
    </Router>
  </BrowserRouter>
</Provider>
`;
